controllers/hosts.go

```diff
@@ -328,6 +328,16 @@ func deleteHostFromNetwork(w http.ResponseWriter, r *http.Request) {
 		// unset all the relayed nodes
 		logic.SetRelayedNodes(false, node.ID.String(), node.RelayedNodes)
 	}
+	if node.IsIngressGateway {
+		// delete ext clients belonging to ingress gateway
+		go func(node models.Node) {
+			if err = logic.DeleteGatewayExtClients(node.ID.String(), node.Network); err != nil {
+				slog.Error("failed to delete extclients", "gatewayid", node.ID.String(), "network", node.Network, "error", err.Error())
+				logic.ReturnErrorResponse(w, r, logic.FormatError(err, "bad request"))
```

Comment:
this will result in two writes to ResponseWriter, one in this goroutine and one in the main function.

one of the writes will be to a response that has already been returned.

Reply:
good catch, will remove it